### ISSUE TITLE
[FEATURE REQUEST] Use multi-threading to speed up benchmark plotting

### ISSUE THREAD

**Issue by yantosca** [geoschem/gcpy#21]
Creating GEOS-Chem output benchmark plots using the run_1mo_benchmark.py script (which calls routines in benchmark.py) can take an extremely long time.  This is especially true for GCHP benchmark plots, which require regridding from cube-sphere to lon/lat.

It would be nice to use Python's multi-threading capabilities to e.g. create all of the PDF files for each category in parallel.  This could drastically reduce the amount of time needed to generate benchmark plots.

**Comment by lizziel** [comment]
I put together something functional using the joblib package a while back that executed plot creation for one collection per thread. It was over a year ago so I used gcpy prototype code in a Jupyter notebook that I had not yet put into the gcpy repo. The non-parallelization part is all very outdated at this point, and I did some extraneous stuff such as storing some input info in a config file. But it did work for parallelization and I was able to create a full suite of collection comparisons within minutes. I am not sure if joblib is the best option, then or now, but here are the files for reference.  

Job submit script:
[make_benchmark_plots.run.txt](https://github.com/geoschem/gcpy/files/3570632/make_benchmark_plots.run.txt)

Script called in job:
[plot_collections.py.txt](https://github.com/geoschem/gcpy/files/3570639/plot_collections.py.txt)

Config file:
[config_bench.py.txt](https://github.com/geoschem/gcpy/files/3570635/config_bench.py.txt)

Script with functions used (can now just use gcpy):
[v1102e_1hr.py.txt](https://github.com/geoschem/gcpy/files/3570638/v1102e_1hr.py.txt)

**Comment by WilliamDowns** [comment]
Parallel functionality has been added to compare_single_level and compare_zonal_mean, where variables are now plotted and merged into a single PDF in parallel. Parallel functionality has also been added to several benchmark functions, where separate PDFs for different categories of variables are created and populated in parallel. These changes have resulted in a 2-4x speedup of benchmark plotting.
